src/main/java/org/jenkinsci/plugins/oic/OicServerWellKnownConfiguration.java

```diff
@@ -0,0 +1,274 @@
+package org.jenkinsci.plugins.oic;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.gson.JsonParseException;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import hudson.Extension;
+import hudson.RelativePath;
+import hudson.Util;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.oic.OicSecurityRealm.TokenAuthMethod;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.verb.POST;
+
+public class OicServerWellKnownConfiguration extends OicServerConfiguration {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Logger LOGGER = Logger.getLogger(OicServerWellKnownConfiguration.class.getName());
+
+    private final String wellKnownOpenIDConfigurationUrl;
+    private String scopesOverride;
+
+    private transient String authorizationServerUrl;
+    private transient String tokenServerUrl;
+    private transient String jwksServerUrl;
+    private transient String endSessionUrl;
+    private transient String scopes;
+    private transient String userInfoServerUrl;
+    private transient boolean useRefreshTokens;
+    private transient TokenAuthMethod tokenAuthMethod;
+
+    /**
+     * Time of the wellknown configuration expiration
+     */
+    private transient LocalDateTime wellKnownExpires = null;
+
+    @DataBoundConstructor
+    public OicServerWellKnownConfiguration(String wellKnownOpenIDConfigurationUrl) {
```

Comment:
JCasC wise you would normally drop the class prefix so this field would be called `openIDConfigurationUrl` as its already in a well known configuration class the well known bit is implied

That way you don't get the double wellKnown in the yaml:
```yaml
securityRealm:
    oic:
      serverConfiguration:
        wellKnown:
          openIDConfigurationUrl: [https://idp.example.com:/someRealm/.well-known/openid-configuration](https://idp.example.com/someRealm/.well-known/openid-configuration)
```

Reply:
is there a way to do this with an annotation so that the config.xml format stays the same?

Reply:
`@Symbol` only works on types (enum, class, interfaces) not on fields or methods, so I will bear this in mind for any future work, but am not planning on adding another data migration at this time.

Reply:
Sure `readResolve` would be the normal approach, there's no alias via annotation.